appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy

```diff
@@ -2,108 +2,113 @@ package nu.yona.server
 
 
 import groovy.json.*
-import spock.lang.Shared
 
-class AddDeviceTest extends AbstractAppServiceIntegrationTest {
-
-	@Shared
-	def richardQuin
-
-	def 'Add user Richard'(){
+class AddDeviceTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Set new device request'()
```

Comment:
That's a matter of taste, I think. The "then" blocks at times contain get operations to validate the "when". This is not reflected in the test name. Otherwise, the test names would become very long. The "then" block however tests everything that was inside the original test.

Your comment might also be particular to the change that I'm making here, as we go from an old test set (with a separate test for "get") to a new one (that uses "get" in the "then" block).